### DISCUSSION TITLE
Cannot upload under specific folder

### DISCUSSION THREAD

**Discussion by cod3rshotout** [Laravel-Backpack/community-forum#1248] [Q&A (Help)]
Hello,

I'm trying to upload the images under "public/uploads" folder, so I've created the following field:

    CRUD::field('gallery')
        ->label(__('backend/installations.gallery'))
        ->type('repeatable')
        ->subfields([
            [
                'name' => 'main_image',
                'label' => __('backend/installations.image'),
                'type' => 'image',
                'withMedia' => [
                    'collection' => 'installation_images',
                    'disk' => 'uploads',
                ]
            ],
        ]);

then, within the model I have configured these two methods:

    public function registerMediaCollections(): void
    {
        $this
            ->addMediaCollection('installation_images')
            ->useDisk('uploads');
    }

    public function registerMediaConversions(?Media $media = null): void
    {
        $this->addMediaConversion('thumb')
            ->width(368)
            ->height(232)
            ->sharpen(10);
    }

I'm using Spatie Media Library, so the model class implement HasMedia.
The `filesystems.php` contains the following code:

    'uploads' => [
        'driver' => 'local',
        'root' => public_path('uploads'),
        'url' => env('APP_URL') . '/uploads',
        'visibility' => 'public',
    ],

the problem's that all the images are uploaded under the storage path, like: "/storage/17/pjZTcwJrnxdk1uGobMLtOM8cXL78gNDaZ8QY4YAw.jpeg"

How is that possible?

[a comment by an automated account is not shown]

**Reply by cod3rshotout** [reply]
I get this error: "array_map(): Argument #2 ($array) must be of type array, null given"

**Comment by cod3rshotout** [comment]
Documentation need to be updated ... it's withFiles not withMedia

**Reply by karandatwani92** [reply]
Hey @cod3rshotout 

Apologies, the **@backpack-ai-bot** answer was wrong but the documentation is correct.
While using `withMedia()`, The folder structure is handled by Spatie's media library.

If you need a custom folder, you can create a separate [filesystem disk](https://laravel.com/docs/11.x/filesystem#configuration) for it and use it.

Or 

You can follow the Spatie's guide to set up a [custom directory structure
](https://spatie.be/docs/laravel-medialibrary/v11/advanced-usage/using-a-custom-directory-structure). If I had to choose, I would go with the first option.
